src/imperator/data/missions.rs

```diff
@@ -57,6 +57,8 @@ impl DbKind for Mission {
             validate_trigger(b, data, &mut sc, Tooltipped::No);
         });
 
+        // TODO - Any scopes that are saved in the on_potential or on_start get added to the ScopeContext of every other block in the mission tree, except for the potential block for the entire tree.
+        // Need to figure out how to propogate these save scopes to the new blocks
```

Comment:
Propagating the saved scopes is done automatically as long as you re-use the same `sc`. 

If you want the potential block for the entire tree to not have those, then you can either put it first or make it its own `sc`.

Reply:
Moving the order should fix the issues with the potential for the whole tree. The main problem though is that MissionTree and MissionTreeTask need to be implemented as seperate Item types and the scopes that are saved in the on_start/on_potential blocks of the MissionTree need to get propogated down to the ScopeContext of all the MissionTreeTask Items that are found in the same file as the MissionTree. Getting the scope context from the MissionTree down to the tasks and only doing it for that file is the main thing I can't figure out here.